src/index.rs

```diff
@@ -1064,6 +1064,53 @@ impl Index {
     Ok(rune_balances)
   }
 
+  pub(crate) fn get_rune_specific_balances(&self, rune: Rune) -> Result<BTreeMap<OutPoint, u128>> {
+    let mut result: BTreeMap<OutPoint, u128> = BTreeMap::new();
+
+    // get rune id
+    let Some(id) = self.database
+      .begin_read()?
+      .open_table(RUNE_TO_RUNE_ID)?
+      .get(rune.0)?
+      .map(|guard| guard.value())
+    else {
+      return Ok(BTreeMap::new());
+    };
+
+    let rune_id = RuneId::load(id);
+    
+    for entry in self
+      .database
+      .begin_read()?
+      .open_table(OUTPOINT_TO_RUNE_BALANCES)?
+      .iter()?
+    {
```

Comment:
In order to make this production ready we would need to add a new table to the index: `RUNE_TO_OUTPOINTS`. 

The way it works right now is that for every request it would iterate through all outpoints, making this a quite expensive call.